medium/users/services.py

```diff
@@ -0,0 +1,38 @@
+from django.contrib.auth.hashers import make_password
+from django.contrib.auth.password_validation import validate_password
+from django.core.validators import validate_email
+
+from medium.users.models import User
+
+
+class UserService:
```

Comment:
I prefer function-based services

Reply:
I prefer class-based services. It is more compact than the separated functions.

Reply:
lets go with class as it reduces code amount.

Reply:
also why we have user servcie class, while some drf view class + serializer can do what is exactly done in this class?

Reply:
We want to accomplish the separation of concern, services will be responsible for the business logic while the views will be responsible for controlling the flow.

Also, I think @OsamaRagab520 will answer this question better than I as it's his idea.

Reply:
you can check out https://github.com/HackSoftware/Django-Styleguide#why-not for more details on why we decided to take DRF's traditional approach